src/mac.rs

```diff
@@ -0,0 +1,206 @@
+use crate::bitfield::BitfieldUnit;
+
+#[repr(C, packed)]
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
+pub struct MacHdr {
+    _bitfield_frame_ctl: BitfieldUnit<[u8; 2]>,
+    duration_id: [u8; 2],
+    address1: [u8; 6],
+    address2: [u8; 6],
+    address3: [u8; 6],
+    sequence_ctl: [u8; 2],
+    address4: [u8; 6],
+    qos_ctl: [u8; 2],
+}
+
+impl MacHdr {
+    pub const LEN: usize = ::core::mem::size_of::<MacHdr>();
+
+    #[inline]
+    pub fn protocol_version(&self) -> u16 {
+        unsafe { ::core::mem::transmute(self._bitfield_frame_ctl.get(0, 2) as u16) }
+    }
+    #[inline]
+    pub fn set_protocol_version(&mut self, val: u16) {
+        unsafe {
+            let val: u16 = ::core::mem::transmute(val);
+            self._bitfield_frame_ctl.set(0, 2, val as u64)
+        }
+    }
+    #[inline]
+    pub fn ty(&self) -> u16 {
+        unsafe { ::core::mem::transmute(self._bitfield_frame_ctl.get(2, 4) as u16) }
+    }
+    #[inline]
+    pub fn set_ty(&mut self, val: u16) {
+        unsafe {
+            let val: u16 = ::core::mem::transmute(val);
+            self._bitfield_frame_ctl.set(2, 4, val as u64)
+        }
+    }
+    #[inline]
+    pub fn sub_ty(&self) -> u16 {
+        unsafe { ::core::mem::transmute(self._bitfield_frame_ctl.get(4, 8) as u16) }
+    }
+    #[inline]
+    pub fn set_sub_ty(&mut self, val: u16) {
+        unsafe {
+            let val: u16 = ::core::mem::transmute(val);
+            self._bitfield_frame_ctl.set(4, 8, val as u64)
+        }
+    }
+    #[inline]
+    pub fn tds(&self) -> u16 {
+        unsafe { ::core::mem::transmute(self._bitfield_frame_ctl.get(8, 9) as u16) }
+    }
+    // set to to_ds
```

Comment:
Can we remove this comment? It isn't really descriptive and it's autogenerated code from bindgen anyways.